tools/resolve-instruction.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+"""
+Prints environment update commands for a specific input instruction. As of now,
+provides configuration for the following which are consistent.
+
+  INTGEMM_CPUID
+  BRT_INSTRUCTION
+  MKL_ENABLE_INSTRUCTIONS
+
+Default operation is autodetect, which resolves to the highest available
+instruction on the hardware run.
+
+Requires the `list_cpu_features` executable from
+https://github.com/google/cpu_features, which provides a cross-platfrom CPU
+detection. The JSON output from `list_cpu_features` is used to resolve the
+highest amongst available instruction on the hardware optionally upto a user
+specified instruction.
+"""
+
+import sys
+import argparse
+import subprocess as sp
+import json
+
+# BRT available instructions (due to INTGEMM), in order of decreasing
+# preference.
+
+available = [
+    'avx512vnni', 
+    'avx512bw',
+    'avx2', 
+    'avx',
+    'ssse3', 
+]
+
+# INTGEMM uses the uppercase of the above.
+INTGEMM_TABLE = { k: k.upper() for k in available }
+
+# MKL has a limitation in SSSE3, where we instead choose SSE4_2.
+
+MKL_TABLE = {
+    "avx512vnni": "AVX512",
+    "avx512bw": "AVX512",
+    "avx2": "AVX2",
+    "avx": "AVX",
+    "ssse3": "SSE4_2",
+}
+
+def env_setup_commands(resolved):
+    # avx codepath is same as SSSE3 for INTGEMM.
+    instruction = "ssse3" if resolved == "avx" else resolved
+
+    envvars = {
+        "INTGEMM_CPUID" : INTGEMM_TABLE[instruction],
+        "BRT_INSTRUCTION" : instruction,
+        "MKL_ENABLE_INSTRUCTIONS" : MKL_TABLE[instruction]
+    }
+    
+    commands = [
+        'export {}={}'.format(key, value) for key, value in envvars.items()
+    ]
+    return ('\n'.join(commands))
+
+def main(args):
+    # Run `list_cpu_features` to obtain JSON output.
+    proc = sp.run(args=[args.path, '--json'], stdout=sp.PIPE, stderr=sp.PIPE)
```

Comment:
So far as I can tell, this is a Rube Goldberg device to get the running CPU model by way of:
- A third party library https://github.com/google/cpu_features#running-sample-code
- Some JSON file
- And some python program that parses the JSON

We already have CPUID, just print out `intgemm::kCPU`  And use a bash case statement to map to environment variables.  

```
// If you want to detect the CPU and dispatch yourself, here's what to use:
enum class CPUType {
  UNSUPPORTED = 0,
  SSE2 = 1,
  SSSE3 = 2,
  AVX2 = 3,
  AVX512BW = 4,
  AVX512VNNI = 5
};
```

Reply:
This change is in effect, except I do not know what to set as MKL vars for SSSE3 and SSE2, which are supported by intgemm, but unable to find an equivalent for MKL. 

There is no information regarding SSSE3 or SSE2 here: https://software.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-fortran/top/support-functions/miscellaneous/mkl-enable-instructions.html.

Docs elsewhere say just change ISA to MKL_ENABLE_INSTRUCTIONS=ISA , but ISA=SSSE3,SSE2 is not available.